source/Octodiff/Core/DeltaApplier.cs

```diff
@@ -24,12 +24,12 @@ public void Apply(Stream basisFileStream, IDeltaReader delta, Stream outputStrea
 
                     var buffer = new byte[4*1024*1024];
                     int read;
-                    long soFar = 0;
-                    while ((read = basisFileStream.Read(buffer, 0, (int)Math.Min(length - soFar, buffer.Length))) > 0)
+                    do
                     {
-                        soFar += read;
+                        read = basisFileStream.Read(buffer, 0, buffer.Length);
+
                         outputStream.Write(buffer, 0, read);
-                    }
+                    } while (read == buffer.Length);
```

Comment:
You should change the `read > buffer.length` to `read > 0` as the stream might return less then what you requested.

https://msdn.microsoft.com/en-us/library/system.io.filestream.read%28v=vs.110%29.aspx
 > An implementation is free to return fewer bytes than requested even if the end of the stream has not been reached.